e_os.h

```diff
@@ -544,6 +544,8 @@ struct servent *getservbyname(const char *name, const char *proto);
 # define CRYPTO_memcmp memcmp
 #endif
 
+#define IF_FAIL_ASSERT(x)     assert(x); if (!(x))
```

Comment:
What is wrong with plain old assert?  I expect @dot-asm will pop up and say why are we building our own layer on top of standard C stuff? :)  And I agree with him here.